lib/go/edgecontext/backend.go

```diff
@@ -0,0 +1,124 @@
+package edgecontext
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	v0 "github.com/reddit/edgecontext/lib/go/internal/v0"
+)
+
+type Backend struct {
+	SetEdgeContext func(context.Context, *EdgeRequestContext) context.Context
+	GetEdgeContext func(context.Context) (*EdgeRequestContext, bool)
+	Factory        func(Config) ecinterface.Factory
+	Init           func(Config) *Impl
+	New            func(context.Context, *Impl, NewArgs) (*EdgeRequestContext, error)
+	FromHeader     func(context.Context, string, *Impl) (*EdgeRequestContext, error)
+}
+
+var defaultBackend = &Backend{
+	SetEdgeContext: func(ctx context.Context, ec *EdgeRequestContext) context.Context {
+		if ec == nil {
+			return ctx
+		}
+		return v0.SetEdgeContext(ctx, ec.Source.(*v0.DataSource))
```

Comment:
:no_bell: I'm _a little_ worried the type assertion here could cause panic

Reply:
The problem is I don't really think we have a good alternative and tbh it probably _should_ panic, if this happens it means you switched the backends while you were handling requests which is a really bad thing to do 😅